shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt

```diff
@@ -568,7 +568,7 @@ object JsonLdUtils {
      */
     fun buildNonReifiedProperty(value: Any): List<Map<String, Any>> =
         listOf(
-            mapOf(JSONLD_ID to value)
+            mapOf(JSONLD_ID to value.toString())
```

Comment:
hmm, any reason?

Reply:
When we compact entity @id wait for a string

Reply:
I know but it should happen automatically. Did you have an error somewhere?

Reply:
Yes, I have an error here [when we serialize datasetId in subjectInfo](https://github.com/stellio-hub/stellio-context-broker/blob/4d01f5e1542f3b4c2655a29cb214a1a1102c1376/search-service/src/main/kotlin/com/egm/stellio/search/authorization/EntityAccessRights.kt#L39)

Reply:
please, elaborate on this:
- what was the exact error?
- when exactly did it happen?

(since this function is used similarly in other places, I need to understand the problem before deciding on a change)

Reply:
Ok I understand.

- The exact error is : 
       `"title": "invalid @id value",
        "detail": "invalid @id value: value of @id must be a string",
        "type": "https://uri.etsi.org/ngsi-ld/errors/BadRequestData"`

- This errors happens during compaction. It comes when we serialize `SubjectRightInfo` at the function `buildNonReifiedProperty` beacause we put a datasetid which is a uri. We associate a uri with`@id` and this is not allowed.

[It is done here](https://github.com/stellio-hub/stellio-context-broker/blob/4d01f5e1542f3b4c2655a29cb214a1a1102c1376/search-service/src/main/kotlin/com/egm/stellio/search/authorization/EntityAccessRights.kt#L39)
That's why I thought parse datasetId when we call the function rather than modify the function directly. 

We can also modify the type of datasetId at the base but I think it is less good

Reply:
if the value of `@id` must be a string, then the `buildNonReifiedProperty` function should only accept a string as argument.

Reply:
`buildExpandedRelationship` accept only an URI and parse this URI to in the function. Maybe we can do the same thing ?